include/dgl/array.h

```diff
@@ -121,6 +148,19 @@ inline bool IsValidIdArray(const dgl::runtime::NDArray& arr) {
   return arr->ndim == 1 && arr->dtype.code == kDLInt;
 }
 
+/*!
+ * \brief Pack a padded tensor to a triplet of concatenated tensor, their lengths,
+ * and offsets.
+ */
+template<typename ValueType>
+std::tuple<NDArray, IdArray, IdArray> Pack(NDArray array, ValueType pad_value);
+
+/*!
+ * \brief Concat array[i, 0:lengths[i]] for every i.  Returns the concatenated array
+ * and offsets in the concatenated array for each original slice.
```

Comment:
Take me a while to understand the docstring. Does it mean that the input `array` must be a 2D matrix? To me, it looks like a batched version of slice. Maybe `BatchSlice` is a more proper name? Also, please write more docstring and perhaps an example.